File: 프로그래머스/0/120852. 소인수분해/소인수분해.swift
import Foundation

func solution(_ n:Int) -> [Int] {
    var answer: [Int] = []
    var num: Int = n
    
    guard n > 1 else { return [] }
    
    let root: Int = Int(Double(num).squareRoot())
    
    for i in 2...max(2, root+1) {
        while num % i == 0 {
            if !answer.contains(i) {
                answer.append(i)
            }
            
            num /= i
        }
    }
    
    if num > 1 {
        answer.append(num)
    }
    
    print(answer)
    
    return answer
}

File: 프로그래머스/0/120852. 소인수분해/소인수분해.py
import math

def solution(n):
    answer = []
    root = math.isqrt(n)
    
    for i in range(2, root + 1):
        while n % i == 0:
            if i not in answer:
                answer.append(i)
            n //= i
    
    if n > 1:
        answer.append(n)
        
    return answer